src/Juno/JunoClient.ts

```diff
@@ -310,6 +374,8 @@ export class JunoClient {
     let data: IGalleryItem;
     if (response.status === HttpStatusCodes.OK) {
       data = await response.json();
+    } else {
+      throw new Error(`HTTP status ${response.status} thrown. ${(await response.json()).Message}`);

```

Comment:
Maybe replace `thrown` by `received`.

Reply:
the only place where this is called is covered by a try catch block , with a notification consle logError message in the catch block.